content/en/samples/posts/2024-12-26-idle-view-app.md

```diff
@@ -0,0 +1,444 @@
+
+- **Operating System**: Windows, Linux, or macOS (tested on Windows 10)
+- **CPU**: No specific requirements (standard modern CPUs are sufficient)
+- **Memory**: 4GB or more
+- **Storage**: At least 256GB of available space
+
+---
+
+## Software Requirements
+This section outlines the essential software required for the project.
+
+- **Node.js**: Required for running the backend server and webOS OSE CLI.
+- **MySQL**: Database management system used in the project.
+- **Prisma ORM**: Used for data modeling and database interactions.
+
+## Version Dependencies in the Project
+This section outlines the essential version dependencies required for the project.
+### Frontend Dependencies
+| Dependency Name       | Current Version | Required/Peer Version(s)     | Notes                          |
+|------------------------|-----------------|------------------------------|--------------------------------|
+| `react`               | ^18.3.1         | Compatible with React ^18.x  | Core library for UI rendering.|
+| `react-dom`           | ^18.3.1         | Matches React version ^18.x  | Required for React rendering. |
+| `vite`                | ^5.4.9          | Node.js >=16.0               | Bundler; needs Node.js v16+.  |
+| `tailwindcss`         | ^3.4.14         | Works with PostCSS ^8.4      | Styling framework.            |
+| `vite-plugin-svgr`    | ^4.3.0          | Compatible with Vite ^5.x    | For SVG imports as components.|
+
+### Backend Dependencies
+| Dependency Name       | Current Version | Required/Peer Version(s)        | Notes                          |
+|------------------------|-----------------|---------------------------------|--------------------------------|
+| `prisma`              | ^5.21.1         | Node.js >=16.x, `@prisma/client` must match | ORM for database.             |
+| `express`             | ^4.21.1         | Works with Node.js >=12.x      | Backend framework.            |
+| `jsonwebtoken`        | ^9.0.2          | Requires `bcryptjs` for hashing| For authentication.           |
+| `mysql2`              | ^3.11.3         | Works with MySQL >=8.x         | Database connector.           |
+
+
+## Project Setup
+This section provides a step-by-step guide to setting up the IdleView project, covering both backend and frontend configurations.
+
+### General Setup
+1. **Clone the repository.**
+   ```sh
+   git clone https://github.com/seangolden11/Idle-View-team7.git
+   cd Idle-View-team7
+   ```
+
+#### Repository Structure
+
+The repository is organized into several directories, each serving a specific purpose:
+
+- **backend**: Contains the server-side code and related resources for the project.
+- **frontend**: Houses the client-side code, including the user interface components and associated assets.
+- **회의록**: (Korean for "meeting records") Contains records of team meetings and discussions.
+- **Project Related Files**: Include miscellaneous files or documents relevant to the project.
```

Comment:
Please remove the descriptions for unnecessary folders for other developers and back up the files before deleting them from the GitHub code.